mmengine/visualization/vis_backend.py

```diff
@@ -816,3 +816,132 @@ def _flatten(self, d, parent_key='', sep='.') -> dict:
             else:
                 items[new_key] = v
         return items
+
+@VISBACKENDS.register_module()
+class ClearmlVisBackend(BaseVisBackend):
+    """Clearml visualization backend class.
+    It requires `clearml`_ to be installed.
+    
+    Examples:
+        >>> from mmengine.visualization import ClearmlVisBackend
+        >>> from mmengine import Config
+        >>> import numpy as np
+        >>> vis_backend = ClearmlVisBackend(save_dir='temp_dir')
+        >>> img = np.random.randint(0, 256, size=(10, 10, 3))
+        >>> vis_backend.add_image('img.png', img)
+        >>> vis_backend.add_scalar('mAP', 0.6)
+        >>> vis_backend.add_scalars({'loss': 0.1,'acc':0.8})
+        >>> cfg = Config(dict(a=1, b=dict(b1=[0, 1])))
+        >>> vis_backend.add_config(cfg)
+
+    Args:
+        save_dir (str): The root directory to save the files
+            produced by the backend.
+        init_kwargs (dict): A dict contains the `clearml.Task.init`
+            initialization keys. See `taskinit`_  for more details.
+        artifact_suffix (Tuple[str] or str, optional): The artifact suffix.
+            Default to ('.py', 'pth').
+            
+    .. _clearml:
+        https://clear.ml/docs/latest/docs/
+    .. _taskinit:
+        https://clear.ml/docs/latest/docs/references/sdk/task/#taskinit
+    """
+    def __init__(self, 
+                 save_dir: str, 
+                 init_kwargs: Optional[dict] = None, 
+                 artifact_suffix: SUFFIX_TYPE = ( '.py', '.pth')):
+        super().__init__(save_dir)
+        self._init_kwargs = init_kwargs
+        self._artifact_suffix = artifact_suffix
+
+    def _init_env(self):
+        if not os.path.exists(self._save_dir):
+            os.makedirs(self._save_dir, exist_ok=True)
```

Comment:
Yes it's not used. 
Currently, there isn't any API to set the directory for cleaml Task.  
If the clearml server is reachable, everything will be logged there. If not, it will be saved in default cache directory like /home/gachiemchiep/.clearml/cache/offline/offline-a9ba36c7bfd24df3ba65d3857687467b.zip